pilot/cmd/pilot-agent/main.go

```diff
@@ -94,7 +94,7 @@ var (
 
 	pilotCertProvider = env.RegisterStringVar("PILOT_CERT_PROVIDER", "istiod",
 		"the provider of Pilot DNS certificate.").Get()
-	jwtPolicy = env.RegisterStringVar("JWT_POLICY", jwt.JWTPolicyThirdPartyJWT,
+	jwtPolicy = env.RegisterStringVar("JWT_POLICY", jwt.PolicyThirdParty,
```

Comment:
env variables are intended for experiments/feature gates - I think it's time to cleanup and add proper mesh config API, and review the use cases and flows in context of multi-revision.

Istiod already supports multiple forms of authentication - it's not a mutually exclusive, but a list that may include client certs, OIDC, JWTs of different kinds from multiple k8s servers.
It is possible that in MC some of the apiservers will use 1p and others 3p JWTs.

The agent should not need any special config - but authenticate with 'provisioning cert' if available, fallback to 3p JWT if available, else 1p JWT. 

For server we'll need a proper config in mesh.config.